File: Xamarin/LightOrganApp.iOS/CustomSearchBar.cs
using System;
using System.Collections.Generic;
using System.Text;
using CoreGraphics;
using UIKit;
using Foundation;

namespace LightOrganApp.iOS
{
    public class CustomSearchBar: UISearchBar
    {
        public CustomSearchBar(CGRect frame): base(frame)
        {
        }

        public override void LayoutSubviews()
        {
            base.LayoutSubviews();
            SetShowsCancelButton(false, false);
        }

        private int? IndexOfSearchFieldInSubviews()
        {
            int? index = null;
            var searchBarView = Subviews[0];

            for (var i = 0; i < searchBarView.Subviews.Length; i++)
            {
                if (searchBarView.Subviews[i] is UITextField)
                {
                    index = i;
                    break;
                }
            }

            return index;
        }

        public override void Draw(CGRect rect)
        {
            var index = IndexOfSearchFieldInSubviews();

            if (index != null)
            {
                var searchField = Subviews[0].Subviews[index.Value] as UITextField;

                searchField.TextColor = UIColor.White;
                searchField.BackgroundColor = UIColor.FromRGB(22, 64, 25);

                var glassIconView = searchField.LeftView as UIImageView;
                if (glassIconView != null)
                {
                    glassIconView.Image = glassIconView.Image?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate);
                    glassIconView.TintColor = UIColor.FromRGB(197, 225, 165);
                }

                var textFieldInsideSearchBarLabel = searchField.ValueForKey(new NSString("placeholderLabel")) as UILabel;
                if (textFieldInsideSearchBarLabel != null)
                    textFieldInsideSearchBarLabel.TextColor = UIColor.FromRGB(197, 225, 165);

                var clearButton = searchField.ValueForKey(new NSString("clearButton")) as UIButton;
                clearButton.SetImage(clearButton.ImageView?.Image?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate), UIControlState.Normal);
                clearButton.SetImage(clearButton.ImageView?.Image?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate), UIControlState.Highlighted);
                clearButton.TintColor = UIColor.FromRGB(197, 225, 165);
            }

            base.Draw(rect);
        }
    }
}

File: iOS10/LightOrganApp/CustomSearchBar.swift
//
//  CustomSearchBar.swift
//  LightOrganApp
//
//  Created by Marcin Kruszyński on 06/05/16.
//  Copyright © 2016 Marcin Kruszyński. All rights reserved.
//

import UIKit

class CustomSearchBar: UISearchBar {
    
    
    override func layoutSubviews() {
        super.layoutSubviews()
        setShowsCancelButton(false, animated: false)
    }
    
    
    func indexOfSearchFieldInSubviews() -> Int! {
        var index: Int!
        let searchBarView = subviews[0]
        
        for i in 0 ..< searchBarView.subviews.count {
            if searchBarView.subviews[i].isKind(of: UITextField.self) {
                index = i
                break
            }
        }
        
        return index
    }
    
    
    override func draw(_ rect: CGRect) {
        
        if let index = indexOfSearchFieldInSubviews() {
            
            let searchField: UITextField = subviews[0].subviews[index] as! UITextField
            
            searchField.textColor = .white
            searchField.backgroundColor = UIColor(red: 0.14, green: 0.14, blue: 0.14, alpha: 1.0)
            
            if let glassIconView = searchField.leftView as? UIImageView {
                glassIconView.image = glassIconView.image?.withRenderingMode(UIImageRenderingMode.alwaysTemplate)
                glassIconView.tintColor = UIColor.white
            }
            
            let textFieldInsideSearchBarLabel = searchField.value(forKey: "placeholderLabel") as? UILabel
            textFieldInsideSearchBarLabel?.textColor = .white
            
            let clearButton = searchField.value(forKey: "clearButton") as! UIButton
            clearButton.setImage(clearButton.imageView?.image?.withRenderingMode(UIImageRenderingMode.alwaysTemplate), for: UIControlState())
            clearButton.setImage(clearButton.imageView?.image?.withRenderingMode(UIImageRenderingMode.alwaysTemplate), for: .highlighted)
            clearButton.tintColor = UIColor.white
        }
        
        super.draw(rect)
    }
}
